src/profiles_management/pmr/classes.py

```diff
@@ -0,0 +1,133 @@
+with them.
+"""
+
+import logging
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Dict, List
+
+import jsonschema
+
+from profiles_management.pmr.schema import RESOURCE_QUOTA_SCHEMA
+
+log = logging.getLogger(__name__)
+
+
+class UserKind(Enum):
+    """Class representing the kind of the user."""
+
+    USER = "user"
+    SERVICE_ACCOUNT = "service-account"
+
+
+class ContributorRole(Enum):
+    """Class representing the role of the user."""
+
+    ADMIN = "admin"
+    EDIT = "edit"
+    VIEW = "view"
+
+
+@dataclass
+class Contributor:
+    """Class representing which users should have access to a Profile."""
+
+    name: str
+    role: ContributorRole
+
+
+@dataclass
+class Owner:
+    """Class representing the owner field of a Profile."""
+
+    name: str
+    kind: UserKind
+
+
+@dataclass
+class Profile:
+    """Class representing a Profile and its Contributors."""
+
+    name: str
+    owner: Owner
+    resources: Dict[str, Any]
+    contributors: List[Contributor]
+
+    # https://docs.python.org/3/library/dataclasses.html#post-init-processing
+    def __post_init__(self):
+        """Validate resourceQuota after values have been initialised."""
+        log.info("Validating ResourceQuota for Profile: %s", self.name)
+        jsonschema.validate(self.resources, RESOURCE_QUOTA_SCHEMA)
+        log.info("ResourceQuota is valid.")
+
+    def has_contributor(self, name: str, role: ContributorRole) -> bool:
+        """Check if the Profile has a contributor with specific role."""
+        for contributor in self.contributors:
+            if contributor.name != name:
+                continue
+
+            if contributor.role != role:
+                continue
+
+            log.info("Profile %s has contributor (%s, %s)", self.name, name, role.value)
+            return True
+
+        log.info("Profile %s doesn't have contributor (%s, %s)", self.name, name, role.value)
+        return False
+
+
+class ProfilesManagementRepresentation:
+    """A class representing the Profiles and Contributors."""
+
+    profiles: dict[str, Profile] = {}
+
+    def __init__(self, profiles: List[Profile] = []):
+        """Initialise based on a list of Profiles.
+
+        If a list of Profiles is given, then the internal dict will be initialised
+        based on this list.
+        """
+        for profile in profiles:
+            self.add_profile(profile)
+
+    def has_profile(self, name: str | None) -> bool:
+        """Check if given Profile name is part of the PMR."""
+        return name in self.profiles
+
+    def add_profile(self, profile: Profile):
+        """Add a Profile to internal dict of Profiles."""
+        self.profiles[profile.name] = profile
+
+    def remove_profile(self, name: str | None):
+        """Remove Prorifle from PMR, if it exists."""
+        if name not in self.profiles:
+            log.info("Profile %s not in PMR.", name)
+            return
+
+        del self.profiles[name]
```

Comment:
In very rare cases, the `del` statement is used in Python code. Perhaps you could replace all of this with:

```python
def remove_profile(self, name: str | None):
  self.profiles.pop(name, 'No key found')
```

Reply:
Sure, but what's the reason for`pop` over `del`?

Reply:
`del` is just less ptyhonic, AFAIK. I'll let you decide.

Reply:
I'd personally keep the `del` and the `log` :)

Reply:
If you are going to keep it, then I think we'll need to add some error handling as `del` will raise a `KeyError` if the key is not in dict.

Reply:
Yep, it's already there